.github/workflows/required-checks-hack-ci.yml

```diff
@@ -0,0 +1,135 @@
+# The jobs in this workflow are necessary because they are configured as
+# required checks on PRs. Github does not currently support configuring required
+# checks per workflow. A PR that only triggers codegen-ci.yml jobs will only
+# report back with the "Codegen Tests" check. But because there is only a global
+# option for required PR checks, that PR would otherwise never satisfy the
+# "Typescript/Python/APIX Tests" checks without these noop jobs.
+#
+# References:
+# https://github.community/t/github-actions-and-required-checks-in-a-monorepo-using-paths-to-limit-execution/16586
+# https://github.community/t/feature-request-conditional-required-checks/16761/6
+#
+# Each job actually has a corresponding "real" job
+# definition in a different workflow file:
+#
+# codegen-ci.yml: Codegen Tests
+# tssdk-ci.yml: Typescript Tests
+# python-ci.yml: Python Tests
+# apix-ci.yml: APIX Tests
+#
+
+name: Required Checks Hack
+on:
+  pull_request:
+
+  push:
+    branches:
+      - main
+
+jobs:
+  satisfy-required-checks:
+    name: Satisfy Required Checks
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repo Checkout
+        uses: actions/checkout@v2
+
+      - uses: tony84727/changed-file-filter@v0.2.0
+        id: filter
+        with:
+          filters: |
+            apix:
+              - packages/code-editor/**
+              - packages/run-it/**
+              - packages/api-explorer/**
+              - packages/extension-api-explorer/**
+              - packages/extension-utils/**
+            codegen:
+              - packages/sdk-codegen/**
+              - packages/sdk-codegen-utils/**
+              - packages/sdk-codegen-scripts/**
+            hackathon:
+              - packages/wholly-sheet/**
+              - packages/hackathon/**
+            python:
+              - python/**
+            resources:
+              - bin/looker-resources-index/**
+              - docs/resources/**
+            tssdk:
+              - packages/sdk/**
+              - packages/sdk-rtl/**
+              - packages/sdk-node/**
+              - packages/extension-sdk/**
+              - packages/extension-sdk-react/**
+              - packages/extension-utils/**
+      - name: Debug info
+        run: |
+          echo "${{ toJSON(steps.filter.outputs) }}"
+          echo "apix: ${{ steps.filter.outputs.apix }}"
+          echo "codegen: ${{ steps.filter.outputs.codegen }}"
+          echo "hackathon: ${{ steps.filter.outputs.hackathon }}"
+          echo "python: ${{ steps.filter.outputs.python }}"
+          echo "resources: ${{ steps.filter.outputs.resources }}"
+          echo "tssdk: ${{ steps.filter.outputs.tssdk }}"
```

Comment:
so we'll need to update this whenever we add another CI workflow. Good to note somewhere

Reply:
Oh, I see you're way ahead of me. Nice.